### PR TITLE
New finished save and load function

### PR DESCRIPTION
**Pull request by LouiBrown**
save and load function works now fine and all conditions are now fulfilled

### CODE REVIEW

**Review by THDMoritzEnderle — CHANGES_REQUESTED** [review]
- Please do not push a saved game. Create the save game folder in the code if it doesnt exist. Do not push the folder
- instead of ls, use os.listdir so it works on windows and linux
- for now, please do not push Dockerfile. This should be a seperate push and will be done by me or jhonny in the futur.
- maybe use .save for the files instead of .txt, is optional
- Please provide docstrings for your functions and check for pylint issues (eg. line too long)
- Remove any comments not meant to be in the final code
- dont push .idea folder

once your  code has been updated, I will look further into the code.
please remember to ONLY push files that are related to the changes. If you want to change anything else unrelated, open a seperate pull requst. Do never push any IDEA files. Use .gitignore to not accidentaly push those files


update: thanks for removing .idea files. Please do so with the docker file too.

**Review by FlorianEder — COMMENTED** [review]
Hab ein paar in Line Kommentare abgegeben, schau die dir mal an (kannst auch gerne darauf antworten wenn ich was falsch verstanden habe)

File: artificial_intelligence.py [hunk 1]
```diff
@@ -41,13 +41,20 @@ def play(self) -> bool:
...
             print(str(self._game_board))
             main_menu.win_menu(self._player_id)
+            print(str(self._game_board))
+            if filename is not None:
+                os.popen(f"rm save_games/{filename}")
             return False
         if self._game_board.check_draw():
             print(str(self._game_board))
             main_menu.draw_menu()
+            print(str(self._game_board))
```
[the remark this answers is not in the document]
Comment (FlorianEder) [reply artificial_intelligence.py:51]: s.u. in diesem Fall, da git komisch ist

File: main.py [hunk 2]
```diff
@@ -2,50 +2,91 @@
...
+        if menu_option == 3:
             break
-
         else:
 
             active_game = True
-            mode = select_gamemode()
+            if gamemode is None:
+                mode = select_gamemode()
+                print(1)
```
[the remark this answers is not in the document]
Comment (FlorianEder) [reply main.py:47]: (bezogen auf print(1))

**Pull request closed without merging**